File: Java/Escola.java
public class Escola {

    private String nome;
    private String endereco;
    private Professor professor;

    public Escola(String nome, String endereco){
        this.nome = nome;
        this.endereco = endereco;

    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEndereco() {
        return endereco;
    }

    public void setEndereco(String endereco) {
        this.endereco = endereco;
    }

    public Professor getProfessor() {
        return professor;
    }

    public void setProfessor(Professor professor) {
        this.professor = professor;
    }

    public void mostrarProfessor(){
        if (professor != null)
            System.out.println("O professor "+professor.getNome()+ " está lecionando na escola "+ this.nome);
        else
            System.out.println(this.nome+" ainda não contratou tal professor. ");

    }





}

File: Python/Escola.py
from Professor import Professor


class Escola:
    def __init__(self, nome, endereco):
        self.nome = nome
        self.endereco = endereco
        self.professor = None

    def get_nome(self):
        return self.nome

    def set_nome(self, nome):
        self.nome = nome

    def get_endereco(self):
        return self.endereco

    def set_endereco(self, endereco):
        self.endereco = endereco

    def get_professor(self):
        return self.professor

    def set_professor(self, professor):
        self.professor = professor

    def mostrar_professor(self):
        if self.professor:
            print(f"O professor {self.professor.get_nome()} está lecionando na escola {self.nome}")
        else:
            print(f"{self.nome} ainda não contratou tal professor.")


if __name__ == "__main__":
    professor = Professor("Carlos", 40)
    escola = Escola("Escola ABC", "Rua XYZ, 123")

    professor.set_escola(escola)
    escola.set_professor(professor)

    professor.mostrar_escola()
    escola.mostrar_professor()
